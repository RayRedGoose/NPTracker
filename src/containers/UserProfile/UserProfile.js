import './UserProfile.scss'
import React from 'react'
import { connect } from 'react-redux'
import logout from 'assets/logout.svg'

export const UserProfile = ({ user }) => {
  return (
    <aside className="user-profile">
      <header>
        <h2>Your Account</h2>
        <img src={logout} alt="logout"/>
      </header>
      <section>
        <div className="image" style={{backgroundImage: `url(${user.image})`}}></div>
        <div>
          <p>{user.name + ' ' + user.last_name}</p>
          <p className="location">{user.city}</p>
        </div>
      </section>
    </aside>
  )
}

export const mapStateToProps = ({ user }) => ({
  user
})

export default connect(mapStateToProps)(UserProfile)
