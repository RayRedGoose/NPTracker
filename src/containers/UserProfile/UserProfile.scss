@import "../../index.scss";

.user-profile {
  padding: 40px 20px;
  @extend %container;
  @extend %active-container;
  @extend %user-show;
  max-width: 360px;
  header {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 40px;
    h2 {
      margin-bottom: 0;
    }
    img {
      cursor: pointer;
      width: 20px;
      margin-left: 20px;
      opacity: 0.5;
      @extend %click-action;
    }
  }
  p {
    font-size: 18px;
    text-align: left;
    &:first-child {
      font-weight: 700;
      letter-spacing: 0.1em;
      margin-top: 0;
    }
    &.location {
      &:before {
        height: 18px;
        margin-right: 5px;
      }
    }
  }
  section {
    display: flex;
    align-items: center;
  }
  .image {
    width: 100px; height: 100px;
    margin-right: 10px;
  }
}
