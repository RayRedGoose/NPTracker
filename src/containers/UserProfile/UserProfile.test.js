import React from 'react'
import { shallow } from 'enzyme'
import { UserProfile, mapStateToProps } from './UserProfile'

describe("UserProfile", () => {
  it("should match snapshot with all data passed correctly", () => {
    const profile = shallow(
      <UserProfile user={{
        name: 'Ray',
        last_name: 'Zlou',
        city: 'Denver',
        image: 'https://www.some.url/some.jpg'
      }}/>
    )
    expect(profile).toMatchSnapshot()
  })
})

describe("mapStateToProps", () => {
  it("should return the object with correct key from store", () => {
    const mockStore = {
      user: {name: 'Ray'},
      activeTab: 2
    }

    const expected = {
      user: {name: 'Ray'}
    }

    const result = mapStateToProps(mockStore)

    expect(result).toEqual(expected)
  })
})
