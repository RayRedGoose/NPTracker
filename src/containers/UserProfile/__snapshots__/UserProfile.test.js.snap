// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match snapshot with all data passed correctly 1`] = `
<aside
  className="user-profile"
>
  <header>
    <h2>
      Your Account
    </h2>
    <img
      alt="logout"
      src="logout.svg"
    />
  </header>
  <section>
    <div
      className="image"
      style={
        Object {
          "backgroundImage": "url(https://www.some.url/some.jpg)",
        }
      }
    />
    <div>
      <p>
        Ray Zlou
      </p>
      <p
        className="location"
      >
        Denver
      </p>
    </div>
  </section>
</aside>
`;
